addons/account/models/partner.py

```diff
@@ -496,6 +502,23 @@ def _compute_bank_count(self):
         for partner in self:
             partner.bank_account_count = mapped_data.get(partner.id, 0)
 
+    def _get_duplicated_bank_accounts(self):
+        self.ensure_one()
+        domains = []
+        for bank in self.bank_ids:
+            domains.append([('acc_number', '=', bank.acc_number), ('bank_id', '=', bank.bank_id.id)])
+        domain = expression.OR(domains)
+        domain = expression.AND([domain, [('partner_id', '!=', self._origin.id)]])
```

Comment:
in case the partner is link to a particular company, I think you should filter the `res.partner.bank` on `(False, partner.company_id.id)`

Reply:
If there is no `bank_ids`, return an empty recordset. It will simplify `_compute_duplicated_bank_account_partners_count` and will prevent a huge number of results in `action_view_partner_with_same_bank` if the method is call due to a wrong `attrs` or something like that